### PR TITLE
make TestUserCredentialsPropertiesOnWindows stable on Windows Server Core

### LINKED ISSUE #46619: Test failure: System.Diagnostics.Tests.ProcessStartInfoTests.TestUserCredentialsPropertiesOnWindows
failed in job: [runtime-libraries-coreclr outerloop 20210105.1 ](https://dev.azure.com/dnceng/public/_build/results?buildId=938368&view=ms.vss-test-web.build-test-results-tab&runId=29665462&resultId=101960&paneView=debug)

net6.0-windows-Release-x64-CoreCLR_release-(Windows.Server.Core.1909.Amd64.Open)windows.10.amd64.server20h1.open@mcr.microsoft.com/dotnet-buildtools/prereqs:windowsservercore-2004-helix-amd64-20200904200251-272704c

Error message
~~~
System.UnauthorizedAccessException : Attempted to perform an unauthorized operation.


Stack trace
   at System.Security.AccessControl.Win32.SetSecurityInfo(ResourceType type, String name, SafeHandle handle, SecurityInfos securityInformation, SecurityIdentifier owner, SecurityIdentifier group, GenericAcl sacl, GenericAcl dacl) in /_/src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/Win32.cs:line 314
   at System.Security.AccessControl.NativeObjectSecurity.Persist(String name, SafeHandle handle, AccessControlSections includeSections, Object exceptionContext) in /_/src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/NativeObjectSecurity.cs:line 263
   at System.Security.AccessControl.NativeObjectSecurity.Persist(String name, AccessControlSections includeSections, Object exceptionContext) in /_/src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/NativeObjectSecurity.cs:line 353
   at System.Security.AccessControl.NativeObjectSecurity.Persist(String name, AccessControlSections includeSections) in /_/src/libraries/System.Security.AccessControl/src/System/Security/AccessControl/NativeObjectSecurity.cs:line 343
   at System.Security.AccessControl.FileSystemSecurity.Persist(String fullPath) in /_/src/libraries/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/FileSystemSecurity.cs:line 124
   at System.IO.FileSystemAclExtensions.SetAccessControl(FileInfo fileInfo, FileSecurity fileSecurity) in /_/src/libraries/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs:line 78
   at System.Diagnostics.Tests.ProcessStartInfoTests.SetAccessControl(String userName, String filePath, Boolean add) in /_/src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs:line 534
   at System.Diagnostics.Tests.ProcessStartInfoTests.TestUserCredentialsPropertiesOnWindows() in /_/src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs:line 503
~~~

### PR DESCRIPTION
**Pull request by adamsitnik**
Fixes #46619

### CODE REVIEW

**Comment by adamsitnik** [comment]
@eiriktsarpalis it's now properly passing

File: src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs [hunk 1]
```diff
@@ -472,8 +472,11 @@ public void TestUserCredentialsPropertiesOnWindows()
             {
                 p = CreateProcessLong();
 
-                // ensure the new user can access the .exe (otherwise you get Access is denied exception)
-                SetAccessControl(username, p.StartInfo.FileName, add: true);
+                if (PlatformDetection.IsNotWindowsServerCore) // for this particular Windows version it fails with Attempted to perform an unauthorized operation (#46619)
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs:475]: The test is still valid without this?  If the test succeeds without this on Windows Server Core, why is it needed on other Windows?

Comment (adamsitnik) [reply src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs:475]: I've added this recently when I was working on re-enabling this test and it was mandatory on my machine to make it work (#46083)

I have no idea why it does not work on machines from this certain pool. Is it a thing related to Windows Server? Some custom helix machine setup?

The test is restricted to admin accounts only:

https://github.com/dotnet/runtime/blob/f6e73ad8c8a3252a5528724fa241fca410295008/src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs#L454

https://github.com/dotnet/runtime/blob/f6e73ad8c8a3252a5528724fa241fca410295008/src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs#L28-L29

Which makes it even more confusing (I would assume that Admin accounts which can create other user accounts can also add permissions to files?)

**Review by danmoseley — APPROVED** [review]
I would like to know the answer too, but merging this gets us into a better state.

**Comment by danmosemsft** [comment]
Hit this again -- so I'm merging!

**Comment by danmosemsft** [comment]
unrelated failures.

**Pull request merged**